api/handlers.go

```diff
@@ -110,3 +111,13 @@ func writeErrorResponse(ctx context.Context, w http.ResponseWriter, msg string,
 	}
 	http.Error(w, msg, status)
 }
+
+func getErrorStatus(err error) int {
+	var status int
+	if errors.Is(err, SessionNotFound) {
```

Comment:
If both statements fail, the status code will be set to 0 (default integer value) this is equivalent to 500 when the http response is interpreted. Which might be what we want but I would suggest to be more explicit, by using something akin to this:
```
...
	switch {
	case errors.Is(err, SessionNotFound):
		status = http.StatusNotFound
	case errors.Is(err, SessionExpired):
		status = http.StatusNotFound
	default:
		status = http.StatusInternalServerError
	}
...
```

Firstly it explicitly sets default status 500 (internal server error) if a case does not match and simplifies future extension of other errors created in the app.

Reply:
This is exactly what I was trying to do in the first place... I was a bit too Java minded about it though by the looks.